test/distributed/org/apache/cassandra/distributed/test/guardrails/GuardrailBulkLoadEnabledTest.java

```diff
@@ -0,0 +1,171 @@
+        assertRows(CLUSTER.get(1).executeInternal("SELECT count(*) FROM bulk_load_tables.test"), row(22L));
+
+        // truncate table
+        truncateGeneratedTables();
+        assertRows(CLUSTER.get(1).executeInternal("SELECT * FROM bulk_load_tables.test"), EMPTY_ROWS);
+
+        // Disable bulk load, stream should fail and no data should be loaded
+        CLUSTER.get(1).runOnInstance(() -> Guardrails.instance.setBulkLoadEnabled(false));
+        long mark = CLUSTER.get(1).logs().mark();
+        tool = ToolRunner.invokeClass(BulkLoader.class,
+                                      "--nodes", NODES,
+                                      "--port", Integer.toString(NATIVE_PORT),
+                                      "--storage-port", Integer.toString(STORAGE_PORT),
+                                      sstablesToUpload.absolutePath());
+
+        CLUSTER.get(1).logs().watchFor(mark, "Bulk load is disabled");
+        tool.asserts().failure().errorContains("Stream failed");
+        assertRows(CLUSTER.get(1).executeInternal("SELECT * FROM bulk_load_tables.test"), EMPTY_ROWS);
+
+        // Enable bulk load again, data should be loaded successfully
+        CLUSTER.get(1).runOnInstance(() -> Guardrails.instance.setBulkLoadEnabled(true));
+        tool = ToolRunner.invokeClass(BulkLoader.class,
+                                      "--nodes", NODES,
+                                      "--port", Integer.toString(NATIVE_PORT),
+                                      "--storage-port", Integer.toString(STORAGE_PORT),
+                                      sstablesToUpload.absolutePath());
+        tool.assertOnCleanExit();
+        assertTrue(tool.getStdout().contains("Summary statistics"));
+        assertRows(CLUSTER.get(1).executeInternal("SELECT count(*) FROM bulk_load_tables.test"), row(22L));
+    }
+
+    private static File prepareSstablesForUpload() throws IOException
+    {
+        generateSstables();
+        File sstable_dir = copySstablesFromDataDir("test");
+        truncateGeneratedTables();
+        return sstable_dir;
+    }
+
+    private static File copySstablesFromDataDir(String table) throws IOException
+    {
+        File cfDir = new File("build/test/cassandra/bulk_load_tables", table);
+        cfDir.tryCreateDirectories();
+        List<String> keyspaceDirPaths = CLUSTER.get(1).callOnInstance(() -> {
+            List<File> cfDirs = Keyspace.open("bulk_load_tables").getColumnFamilyStore(table).getDirectories().getCFDirectories();
+            return transform(cfDirs, (d) -> d.absolutePath());
+        });
+        for (File srcDir : transform(keyspaceDirPaths, (p) -> new File(p)))
+        {
+            for (File file : srcDir.tryList((file) -> file.isFile()))
+            {
+                FileUtils.copyFileToDirectory(file.toJavaIOFile(), cfDir.toJavaIOFile());
+            }
+        }
+        return cfDir;
+    }
+
+    private static void generateSstables() throws IOException
```

Comment:
This method actually does not throw `IOException`. It does not need to throw anything.